plugins/catalog-backend-module-github/src/providers/GithubEntityProvider.ts

```diff
@@ -242,6 +267,200 @@ export class GithubEntityProvider implements EntityProvider {
       presence: 'optional',
     };
   }
+
+  /** {@inheritdoc @backstage/plugin-events-node#EventSubscriber.onEvent} */
+  async onEvent(params: EventParams): Promise<void> {
+    this.logger.debug(`Received event from ${params.topic}`);
+    if (params.topic !== TOPIC_REPO_PUSH) {
+      return;
+    }
+
+    if (params.metadata?.['x-github-event'] === 'push') {
```

Comment:
I guess you don't really need this check anymore as this is provided by the topic you subscribed to already. And there is even the topic check at the topic.

I removed this at another PR for the Bitbucket Cloud implementation, too. Was a leftover from a previous iterations there.